src/components/LinkChooser/types.ts

```diff
@@ -0,0 +1,134 @@
+        context: LinkChooserContext;
+    }
+>;
+
+export type TemplateMenuBlock = {
+    id: string;
+    menuItems: SearchResult[];
+    ariaLabel?: string;
+};
+
+export type LinkChooserProps = {
+    openInNewTab: boolean;
+    ariaLabel?: string;
+    label?: string;
+    placeholder?: string;
+    onOpenInNewTabChange: (value: boolean) => void;
+    onLinkChange: (value: SearchResult | null) => void;
+    readonly copyToClipboard: Clipboard;
+    readonly getGlobalByQuery?: (query: string) => Promise<SearchResult[]>;
+    readonly getGuidelinesByQuery?: (query: string) => Promise<SearchResult[]>;
+    readonly getTemplatesByQuery?: (query: string) => Promise<SearchResult[]>;
+    readonly openPreview: (value: string, target: string) => void;
+};
+
+export type SearchResult = Omit<MenuItemType, "title"> & { icon: string; title: string };
+
+export type SearchResultListProps = AriaListBoxOptions<unknown> & {
+    listBoxRef?: RefObject<HTMLUListElement>;
+    state: ListState<unknown>;
+    menuBlocks: (Omit<MenuBlock, "menuItems"> & { menuItems: SearchResult[] })[];
+    noBorder?: boolean;
+    hasItems?: boolean;
+    query: string;
+    machineService: MachineService;
+};
+
+export type SearchResultSectionProps = {
+    heading: Node<unknown>;
+    state: ListState<unknown>;
+    keyItemRecord: Record<string, SearchResult>;
+    machineService: MachineService;
+};
+
+export type SearchResultOptionProps = {
+    item: Node<unknown>;
+    state: ListState<unknown>;
+    keyItemRecord: Record<string, SearchResult>;
+    machineService: MachineService;
+};
+
+export type TemplateProps = {
+    title: ReactNode;
+    subtitle?: string;
+    preview?: string;
+};
+
+export type SectionActionMenuProps = {
+    machineService: MachineService;
+};
+
+export type SearchInputProps = {
+    id?: string;
+    selectedResult: SearchResult | null;
+    type?: HTMLInputTypeAttribute;
+    decorator?: ReactElement;
+    dotted?: boolean;
+    previewable?: boolean;
+    copyable?: boolean;
+    clearable?: boolean;
+    placeholder?: string;
+    required?: boolean;
+    disabled?: boolean;
+    validation?: Validation;
+    value?: string | number | readonly string[];
+    onChange?: ChangeEventHandler<HTMLInputElement> | undefined;
+    onEnterPressed?: (event: KeyboardEvent<HTMLInputElement>) => void;
+    onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+    onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+    onClear?: () => void;
+    onPreview?: () => void;
+    onClick?: () => void;
+    size?: number;
+    machineService: MachineService;
+};
+
+export type PopoverProps = {
+    popoverRef?: React.RefObject<HTMLDivElement>;
+    children: React.ReactNode;
+    isOpen?: boolean;
+    onClose: () => void;
+};
+
+export type LinkChooserContext = {
+    searchResults: SearchResult[];
+    selectedResult: SearchResult | null;
+    query: string;
+    interruptedFetch: boolean;
+    readonly copyToClipboard: Clipboard;
```

Comment:
with the given naming, this seems to be a function

Reply:
done 👍